### ISSUE TITLE
App taking 5/6 seconds before starting up - analytics problem?

### ISSUE THREAD

**Issue by yesterday24** [invertase/react-native-firebase#5014]
My app is taking a long time from start up to resolving the bootsplash. I'm using react-native-bootsplash.

I've installed some RN-firebase components (analytics, crashlytics, in-app-messaging, messaging, storage, remote-config). I already deleted in-app-messaging and remote-config to check if the problem was with one of these (because these were the last ones I installed).

When running in **dev** mode in an IOS simulator or real device (same behaviour), it takes up to 5-6 seconds before the app starts working (after the green bar with the "loading javascript bundle 100%" disappears).

I've added some xcode output:

```
...
2021-03-11 15:57:56.834740+0100 myapp[14883:113320] 7.6.0 - [Firebase/Analytics][I-ACS031006] View controller already tracked. Class, ID: UIViewController, -89530970983276176
2021-03-11 15:57:56.964632+0100 myapp[14883:113328] [connection] nw_socket_handle_socket_event [C8:1] Socket SO_ERROR [61: Connection refused]
2021-03-11 15:57:56.965042+0100 myapp[14883:113328] [connection] nw_connection_get_connected_socket [C8] Client called nw_connection_get_connected_socket on unconnected nw_connection
[*] 2021-03-11 15:57:56.965344+0100 myapp[14883:113328] TCP Conn 0x60000070a260 Failed : error 0:61 [61]
[*] 2021-03-11 15:58:01.619399+0100 myapp[14883:113326] 7.6.0 - [Firebase/Analytics][I-ACS023012] Analytics collection enabled
2021-03-11 15:58:01.619548+0100 myapp[14883:113326] 7.6.0 - [Firebase/Analytics][I-ACS023220] Analytics screen reporting is enabled. Call +[FIRAnalytics logEventWithName:FIREventScreenView parameters:] to log a screen view event. To disable automatic screen reporting, set the flag FirebaseAutomaticScreenReportingEnabled to NO (boolean) in the Info.plist
2021-03-11 15:58:01.902822+0100 myapp[14883:113230] [native] [GESTURE HANDLER] Initialize gesture handler for root view <RCTRootContentView: 0x7fbbc5913e20; reactTag: 1; frame = (0 0; 414 896); gestureRecognizers = <NSArray: 0x600003e470c0>; layer = <CALayer: 0x600003076260>>
2021-03-11 15:58:02.120608+0100 myapp[14883:113442] 7.6.0 - [Firebase/Analytics][I-ACS031029] Logging screen view with screen name and screen class: Authentication, Authentication
...
```

I have added [*] to the two lines I think are important: the first line has datetime 15:57:56, the second line has datetime 15:58:01, so there are almost 5 seconds between these two lines where the app is doing nothing. When I disable analytics I still get the first output line (TCP Conn failed...). I don't know if it has anything to do with RN-firebase but because it appears between the output lines of RN-firebase I'm guessing it has something to do with it (analytics perhaps?).

I've checked [this issue](https://github.com/facebook/react-native/issues/21030) but after implementing the "solution" posted there, the startup time of the app is still +/- 5 seconds, so no change at all.

In my feeling, it looks like that firebase-analytics is taking some time to start up. Because when retesting the app, it's always the second line [*] that pops up after 5/6 seconds (analytics collection enabled).

I have a firebase.json-file with ``"analytics_auto_collection_enabled": false`` and I manually enable the analytics connection in a useEffect()-function in App.js.

**Comment by mikehardy** [comment]
Did you remove analytics and try again? That would be the next reasonable test I think.
If that did it, then I would try with a version override of firebase-ios-sdk to newly released 7.8.0 to see if current fixes it
You might scan that repo's issue list

You should definitely follow the issue template though, you did not, so we can't really help
